cmd/api-server/cli/root.go

```diff
@@ -60,7 +60,17 @@ func NewCommandStartComposeServer(stopCh <-chan struct{}) *cobra.Command {
 			if err := utilerrors.NewAggregate(errors); err != nil {
 				return err
 			}
-			return runComposeServer(o, stopCh)
+			for {
+				err := runComposeServer(o, stopCh)
+				// If the compose-api starts before the API server is listening then we can get a transient connection refused
+				// while looking up missing authentication information in the cluster (see #120).
+				if err != nil && strings.HasSuffix(err.Error(), "connection refused") {
```

Comment:
Depending on error content is somehow fragile, even if I guess there's no other way to catch this one... Maybe using `contains` would be a "little" more robust on changes ?

Reply:
Good point. I've pushed a patch switching to `Contains` instead of `HasSuffix`.

I had a go at doing something more precise yesterday to see what it would look like. From memory I think you have to do something like
```golang
if urlError, ok := err.(*url.Error); ok {
  if opError, ok := urlError.Err.(*net.OpError); ok {
     if syscallError, ok := opError.Err.(*os.SyscallError); ok {
       // at this point it probably has "connect" in it
     }
  }
}
```
Some of those error types have `Temporary()` functions which I hoped I could use but they all returned `false` when I tried it.
Let me know what you prefer! :)

Reply:
I'm good with the `contains` 😄